src/Mvc/Mvc.Core/src/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -138,5 +190,50 @@ private static void InferContentTypes(ActionContext context, ObjectResult result
                 result.ContentTypes.Add("application/problem+xml");
             }
         }
+
+        private Task<object> EnumerateAsyncEnumerable(IAsyncEnumerable<object> value)
+        {
+            var type = value.GetType();
+            if (!_asyncEnumerableConverters.TryGetValue(type, out var result))
+            {
+                var enumerableType = ClosedGenericMatcher.ExtractGenericInterface(type, typeof(IAsyncEnumerable<>));
+                result = null;
+                if (enumerableType != null)
+                {
+                    var enumeratedObjectType = enumerableType.GetGenericArguments()[0];
```

Comment:
Btw, we need to do this because XML serializer needs to be told the type of the list. Giving it a vanilla `List<object>` causes them to not be serialized correctly.

Reply:
I should probably add a note